CONTRIBUTING.md

```diff
@@ -0,0 +1,197 @@
+
+## Building the code
+
+Inspektor Gadget is composed by a client executable and a container image.
+A container repository is needed to push the image. The following commands
+use the value of the `CONTAINER_REPO` env variable, it defaults to
+`docker.io/kinvolk/gadget` if not defined.
+
+### Building the client executable
+
+You can compile for all supported platforms by running `make kubectl-gadget`
+or build for a specific one with `make kubectl-gadget-linux-amd64` or `make kubectl-gadget-darwin-amd64`.
+
+Note:
+- The compilation uses `tools/image-tag` to choose the tag of the container
+image to use according to the branch that you are compiling.
+- The container repository is set with the `CONTAINER_REPO` env variable.
+- You can push the container images to another registry and use the `--image`
+argument when deploying to the Kuberentes cluster.
+- If you wish to make changes to traceloop program, update
+`gadget-container/gadget.Dockerfile` to pick your own image of traceloop.
+
+See the [minikube](#Development-environment-on-minikube-for-the-traceloop-gadget)
+section for a faster development cycle.
+
+### Building the gadget container image
+
+You can build and push the container gadget image by running the following commands:
+
+```
+$ cd gadget-container
+$ make build
+$ make push
+```
+
+The BPF code is built using a Docker container, so you don't have to worry
+installing the compilers to build it.
+
+## Workflows
+
+### Github Actions
+
+This repo uses Github actions as CI. It compiles and uploads the Inspektor Gadget
+executable and gadget container image. It also runs unit and some integration tests.
+A fork of this project should
+enable them in the repo settings page and add the following
+[secrets](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository) to be able to use them:
+
+- `CONTAINER_REPO`: The container repository to use. Example: docker.io/kinvolk/gadget
+- `CONTAINER_REGISTRY_USERNAME` & `CONTAINER_REGISTRY_PASSWORD`: Authentication information for the the repo above.
+
+### Development environment on minikube for the traceloop gadget
+
+It's possible to make changes to traceloop and test them on minikube locally without pushing container images to any registry.
+
+* Make sure the git repositories `traceloop` and `inspektor-gadget` are cloned in sibling directories
+* Minikube with the Docker driver does not work for traceloop. You can use another driver, for example:
+```
+$ minikube start --driver=kvm2
+```
+* Install Inspektor Gadget on minikube as usual:
+```
+$ kubectl gadget deploy | kubectl apply -f -
+```
+* Make changes in the traceloop repository and compile with `make`
+* Generate the new gadget image and deploy it to minikube:
+```
+$ make -C gadget-container/ minikube
+```
+
+Note that the minikube image only works with the traceloop gadget.
```

Comment:
This should no longer be true after https://github.com/kinvolk/inspektor-gadget/pull/113